paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java

```diff
@@ -116,6 +116,38 @@ public static <T extends Entity> CraftEntity getEntity(CraftServer server, T ent
         throw new AssertionError("Unknown entity " + (entity == null ? null : entity.getClass()));
     }
 
+    public Entity getHandle() {
+        return this.entity;
+    }
+
+    public Entity getHandleRaw() {
+        return this.entity;
+    }
+
+    public void setHandle(final Entity entity) {
+        this.entity = entity;
+    }
+
+    @Override
+    public String toString() {
+        return this.getClass().getSimpleName() + "{uuid=" + this.getUniqueId() + '}';
```

Comment:
Maybe can just use handle.getScoreboardName so we don't have to leave the toString implementation in CraftPlayer either.